modules/planning/reference_line/reference_line.cc

```diff
@@ -409,18 +409,21 @@ bool ReferenceLine::IsOnRoad(const SLPoint& sl_point) const {
 // return a rough approximated SLBoundary using box length. It is guaranteed to
 // be larger than the accurate SL boundary.
 bool ReferenceLine::GetApproximateSLBoundary(
-    const common::math::Box2d& box, SLBoundary* const sl_boundary) const {
-  SLPoint center_sl_point;
-  if (!XYToSL(box.center(), &center_sl_point)) {
-    AERROR << "failed to get projection for point: "
-           << box.center().DebugString() << " on reference line.";
+    const common::math::Box2d& box, const double start_s, const double end_s,
+    SLBoundary* const sl_boundary) const {
+  double s = 0.0;
+  double l = 0.0;
+  double distance = 0.0;
+  if (!map_path_.GetProjectionWithHueristicParams(box.center(), start_s, end_s,
+                                                  &s, &l, &distance)) {
+    AERROR << "Can't get projection point from path.";
     return false;
   }
 
-  sl_boundary->set_start_s(center_sl_point.s() - box.length() / 2.0);
-  sl_boundary->set_end_s(center_sl_point.s() + box.length() / 2.0);
-  sl_boundary->set_start_l(center_sl_point.l() - box.length() / 2.0);
-  sl_boundary->set_end_l(center_sl_point.l() + box.length() / 2.0);
+  sl_boundary->set_start_s(s - box.length() / 2.0);
+  sl_boundary->set_end_s(s + box.length() / 2.0);
+  sl_boundary->set_start_l(l - box.length() / 2.0);
```

Comment:
box.width()?

Reply:
or you can use box's heading to calculate the exact start_s, end_s, etc.

Reply:
It should be length because we need a box that can guarantee to cover thr actual box. 
We are unable to calculate the accurate start_s and end_s from he box’s heading, we also need to know the heading of the projected point on ref line. But it will double the calculation time.